src/opening-period/form/OpeningPeriodForm.tsx

```diff
@@ -0,0 +1,322 @@
+import React, { useState } from 'react';
+import { ArrayField, useFieldArray, useForm } from 'react-hook-form';
+import { useHistory } from 'react-router-dom';
+import {
+  DatePeriod,
+  DatePeriodOptions,
+  FormWeekdays,
+  Language,
+  ResourceState,
+  ResourceStateApiOption,
+  ResourceStateOption,
+  TimeSpan as TimeSpanApiFormat,
+  TimeSpanFormFormat,
+} from '../../common/lib/types';
+import { transformToApiFormat } from '../../common/utils/date-time/format';
+import Datepicker from '../../components/datepicker/Datepicker';
+import { PrimaryButton, SecondaryButton } from '../../components/button/Button';
+import { ErrorToast, SuccessToast } from '../../components/notification/Toast';
+import OpeningPeriodDescription from '../description/OpeningPeriodDescription';
+import TimeSpan from '../time-span/TimeSpan';
+import './OpeningPeriodForm.scss';
+
+type OpeningPeriodDataBase = {
+  openingPeriodTitle: string;
+  openingPeriodOptionalDescription: string;
+  openingPeriodBeginDate: string | undefined;
+  openingPeriodEndDate: string | undefined;
+};
+
+interface OpeningPeriodFormData extends OpeningPeriodDataBase {
+  timeSpans: Array<TimeSpanFormFormat> | {}[];
+}
+
+export interface OpeningPeriodApiData extends OpeningPeriodDataBase {
+  timeSpans: Array<TimeSpanApiFormat>;
```

Comment:
I think in api data the names are in snake_case? So I wonder why timeSpans is in camelCase here?

Reply:
Good question, I check this one 🤔

Reply:
This was completely redundant type declaration.

Reply:
Removed declaration